front/components/data_source/CreateConnectionConfirmationModal.tsx

```diff
@@ -24,6 +25,15 @@ export function CreateConnectionConfirmationModal({
   onConfirm,
 }: CreateConnectionConfirmationModalProps) {
   const [isLoading, setIsLoading] = useState(false);
+  const [extraConfig, setExtraConfig] = useState<string | null>(null);
+
+  const isExtraConfigValid = useCallback(() => {
+    if (connectorProviderConfiguration.connectorProvider === "zendesk") {
+      return /^[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?$/.test(extraConfig || "");
+    } else {
+      return true;
+    }
+  }, [connectorProviderConfiguration.connectorProvider, extraConfig]);
```

Comment:
Not ideal, as this function will be re-callbackify on each key stroke in `extraConfig`. Instead, you should pass `extraConfig` as a function param.

Reply:
sure thing